llvm/lib/Target/AMDGPU/AMDGPUPromoteAlloca.cpp

```diff
@@ -771,7 +771,8 @@ bool AMDGPUPromoteAllocaImpl::tryPromoteAllocaToVector(AllocaInst &Alloca) {
     }
 
     // Ignore assume-like intrinsics and comparisons used in assumes.
-    if (isAssumeLikeIntrinsic(Inst)) {
+    // Make sure that intrinsics do not have any use like e.g. llvm.objectsize
+    if (isAssumeLikeIntrinsic(Inst) && Inst->use_empty()) {
```

Comment:
This change will fix the crash, but I'm not sure about it overall.
We have at least three intrinsics from `isAssumeLikeIntrinsic` which are returning something.
`objectsize`, `invariant.start` and `ptr.annotation`. Maybe we should allow `objectsize` to pass this check and add `objectsize` to the `WorkList` and later in `promoteAllocaUserToVector` we could just calculate the size of the alloca ?. If IR contains `invariant.start` or `ptr.annotation` we will just quit.

Reply:
Instead of this you should do 
```
if(isAssumeLikeIntrinsic(Inst)) {
  if(!Inst->use_empty()) {
        RejectUserr(Inst, "assume-like intrinsic cannot have any users");
        return false;
  }
```
To explicitly reject them IMO.